src/builders/rollup/build.ts

```diff
@@ -83,7 +83,15 @@ export async function rollupBuild(ctx: BuildContext): Promise<void> {
       dts(ctx.options.rollup.dts),
       removeShebangPlugin(),
       ctx.options.rollup.emitCJS && fixCJSExportTypePlugin(),
-    ].filter(Boolean);
+    ].filter(
+      (plugin) =>
+        /**
+         * Issue: #396
+         * rollup-plugin-dts conflicts with rollup-plugin-commonjs:
+         * https://github.com/Swatinem/rollup-plugin-dts?tab=readme-ov-file#what-to-expect
+         */
+        plugin && (!("name" in plugin) || plugin.name !== "commonjs"),
```

Comment:
Why we need to _also_ filter plugins without name here?

Reply:
I think you might have misunderstood. It means plugins with either _"no name"_ or _"a name, but the name is not equal to 'commonjs'"_ are kept.

(I'm not using `plugin?.name !== "commonjs"` because that would cause a ts error 😅)